R/gradethis_exercise_checker.R

```diff
@@ -104,7 +104,13 @@ check_exercise <- function(
 
   # Copy over all learnr args into the checking environment
   for (name in names(learnr_args)) {
-    check_obj_envir[[paste0(".", name)]] <- learnr_args[[name]]
+    # Ensure that code objects are always a length-1 character string
+    transform <- identity
+    if (grepl("code", name) && is.character(learnr_args[[name]])) {
+      transform <- function(x) paste(x, collapse = "\n")
+    }
+    
+    check_obj_envir[[paste0(".", name)]] <- transform(learnr_args[[name]])
```

Comment:
I think altering a variable in place is a lil less complex. Added a check to only perform on > length 1 vectors

```suggestion
    learnr_arg <- learnr_args[[name]]
    if (length(learnr_arg) > 1 && grepl("code", name) && is.character(learnr_args[[name]])) {
      learnr_arg <- paste(learnr_arg, collapse = "\n")
    }
    
    check_obj_envir[[paste0(".", name)]] <- learnr_arg
```

Reply:
yeah that's a really good point, thanks